fuse_models/src/unicycle_2d_ignition.cpp

```diff
@@ -88,6 +88,11 @@ void Unicycle2DIgnition::onInit()
   if (!params_.reset_service.empty())
   {
     reset_client_ = node_handle_.serviceClient<std_srvs::Empty>(ros::names::resolve(params_.reset_service));
+
+    while (!reset_client_.waitForExistence(ros::Duration(30.0)) && ros::ok())
```

Comment:
The `30.0`s timeout dictates the `ROS_WARN_STREAM` print rate. Maybe it could be set to a lower value like `10.0`.

The check for `ros::ok()` is because `waitForExistence` could also `return false` if the node has been shutdown: http://docs.ros.org/indigo/api/roscpp/html/classros_1_1ServiceClient.html#a530e3f1d55cf50ab00b8e5bbd9e8230a

Reply:
This actually doesn't work because it's called before the `reset` service is advertised in the `FixedLagSmoother` constructor, and happens in the base class `Optimizer` constructor, so it creates some sort of "deadlock". I'm trying other places, but it looks like it's not possible to guarantee the service is available yet in the init/start methods. I guess worst case I could add a check in the `process`  method.

Reply:
I've updated the PR to wait for the `reset` service in the `process` method, right before the service client is called.

Reply:
Yeah. I have a future task for myself to rework the optimizer construction sequence. I feel like we should completely initialize the derived optimizer before loading the plugins. But with the current inheritance scheme, that is not really possible.